### ISSUE TITLE
Sublime-gulp should respect "folder_exclude_patterns" and "file_exclude_patterns"

### ISSUE THREAD

**Issue by Ivanca** [NicoSantangelo/sublime-gulp#79]
Right now when I execute the gulp command it asks me which of 2 `gulpfile.js` should I use, despite one of them being excluded from the project (the file with extension `.sublime-project`) which makes no sense.

More info about "folder_exclude_patterns" and "file_exclude_patterns" see https://www.sublimetext.com/docs/3/projects.html

**Comment by NicoSantangelo** [comment]
The problem with using `folder_exclude_patterns` and `file_exclude_patterns` is that if a user in the same situation won't be able to use that `gulpfile` even if it wants to. That's why (early on) I've added the `ignored_gulpfile_folders` setting.

That said, we could add a boolean setting to append `folder_exclude_patterns` and `file_exclude_patterns` to `ignored_gulpfile_folders` defaulting to false (to keep current behaviour and don't break existing workflows)

What do you think?

**Comment by Ivanca** [comment]
`folder_exclude_patterns` and `file_exclude_patterns` are established by the Sublime Text user; so basically you are asking what happens if the user establishes a configuration and later on the user himself regrets it? That makes little sense to me; anyway, one can always change those options to make the gulpfile appear again.

But arguably this is a problem with Sublime itself, it shouldn't even give plugins access to files outside the project.

Of course `ignored_gulpfile_folders` is better than nothing and I'm glad it exists.

**Comment by Ivanca** [comment]
`ignored_gulpfile_folders` its a global option and not a per-project option? ... *facepalm*

**Comment by Ivanca** [comment]
Nope; I can't make "ignored_gulpfile_folders" to work, apparently doing `"ignored_gulpfile_folders": ["app"]` it's not enough to make it stop looking for gulpfiles inside folders called `app`. 

Another reason to not implement your own options when possible.
